packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,221 @@
+
+    /// @notice DeputyGuardianModule used by the Security Council Safe.
+    IDeputyGuardianModule internal immutable DEPUTY_GUARDIAN_MODULE;
+
+    /// @notice Address of the deputy account.
+    address internal immutable DEPUTY;
+
+    /// @notice Maximum signature validity in seconds.
+    uint256 internal immutable MAX_SIGNATURE_VALIDITY_SECONDS;
+
+    /// @notice Typehash for the Pause action.
+    bytes32 internal constant PAUSE_MESSAGE_TYPEHASH = keccak256("PauseMessage(uint256 expiry,bytes32 nonce)");
+
+    /// @notice Typehash for the DeputyAuth message.
+    bytes32 internal constant DEPUTY_AUTH_MESSAGE_TYPEHASH = keccak256("DeputyAuthMessage(address deputy)");
+
+    /// @notice Used nonces.
+    mapping(bytes32 => bool) public usedNonces;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @param _foundationSafe Address of the Foundation Safe.
+    /// @param _deputyGuardianModule Address of the DeputyGuardianModule used by the SC Safe.
+    /// @param _maxSignatureValiditySeconds Maximum signature validity in seconds.
+    /// @param _deputy Address of the deputy account.
+    /// @param _deputySignature Signature from the deputy verifying that the account is an EOA.
+    constructor(
+        Safe _foundationSafe,
+        IDeputyGuardianModule _deputyGuardianModule,
+        uint256 _maxSignatureValiditySeconds,
+        address _deputy,
+        bytes memory _deputySignature
+    )
+        EIP712()
+    {
+        // Check that the deputy is an EOA.
+        // We do not support EIP-1271 signatures here.
+        bytes32 digest = _hashTypedData(keccak256(abi.encode(DEPUTY_AUTH_MESSAGE_TYPEHASH, DeputyAuthMessage(_deputy))));
+        if (ECDSA.recover(digest, _deputySignature) != _deputy) {
+            revert DeputyPauseModule_InvalidDeputy();
+        }
+
+        // Zero second validity would mean that signatures are never valid.
+        if (_maxSignatureValiditySeconds == 0) {
+            revert DeputyPauseModule_SignatureValidityTooShort();
+        }
+
+        // Constant 1 year maximum validity. 1 year is a sane default and guarantees that the
+        // signature validity will never be excessively long. We accept that this slightly limits
+        // the configurability of this module, but it's not a module that's meant to be deployed
+        // on arbitrary Safe contracts, only the Foundation Safe.
+        if (_maxSignatureValiditySeconds > 365 days) {
+            revert DeputyPauseModule_SignatureValidityTooLong();
+        }
+
+        // Initialize the immutable variables.
+        // We do not perform zero address checks here and expect the deployer to perform
+        // appropriate validation on these values.
+        FOUNDATION_SAFE = _foundationSafe;
+        DEPUTY_GUARDIAN_MODULE = _deputyGuardianModule;
+        DEPUTY = _deputy;
+        MAX_SIGNATURE_VALIDITY_SECONDS = _maxSignatureValiditySeconds;
```

Comment:
We could get a bit more robust and ensure the foundation safe is actually a safe by calling a getter, similar for the deputyGuardianModule

Reply:
Good point, might as well do that.

Reply:
Same as https://github.com/ethereum-optimism/optimism/pull/13186#discussion_r1868490027, let's just track the semgrep rule idea somewhere